internal/services/storage/storage_account_resource.go

```diff
@@ -2275,6 +2436,23 @@ func resourceStorageAccountDelete(d *pluginsdk.ResourceData, meta interface{}) e
 	return nil
 }
 
+func handleStorageServiceError(service string, operation string, err error) (interface{}, string, error) {
+	// if the error is a DetailedError type, check the status code and return the appropriate state
+	var respErr azautorest.DetailedError
+	if errors.As(err, &respErr) {
+		log.Printf("[DEBUG] error %s %s service properties: statusCode=%d, message=%s, error=%s\n", operation, service, respErr.StatusCode.(int), respErr.Message, err)
+		// if the status code is 404 (not found), retry the request
+		if response.WasNotFound(respErr.Response) {
+			// if the status code is 404 (not found), retry the request
+			log.Printf("[DEBUG] %s service properties not available, retrying...\n", service)
+			return false, "Pending", nil
+		}
+	}
+	// if the error is unhandled type or status, log the error and return the error state
+	log.Printf("[DEBUG] unexpected error while %s %s service properties: %v\n", operation, service, err)
```

Comment:
I think this should be `%+v` instead of `%v` to expand the underlying error struct?

```suggestion
	log.Printf("[DEBUG] unexpected error while %s %s service properties: %+v\n", operation, service, err)
```